libs/SalesforceSwiftSDK/SalesforceSwiftSDK/Extensions/SFSmartSyncExtensions.swift

```diff
@@ -0,0 +1,260 @@
+            }
+        }
+
+        /**
+         Check to see of Sync exists
+         
+         ```
+         syncManager.Promises.hasSync(name: syncName)
+         .then { result in
+         ..
+         }
+         ```
+         - parameter name: Name of sync.
+         - Returns: Boolean result wrapped in a promise.
+         */
+        func hasSync(name: String) -> Promise<Bool> {
+            return Promise(.pending) {  resolver in
+                resolver.fulfill(self.api!.hasSync(withName: name))
+            }
+        }
+        
+        /**
+         Delete a sync
+         
+         ```
+         syncManager.Promises.deleteSync(syncId:  id)
+         .then {
+         ..
+         }
+         ```
+         - parameter name: Name of sync.
+         */
+        func deleteSync(syncId: NSNumber) -> Promise<Void>  {
+            return Promise(.pending) {  resolver in
+                resolver.fulfill(self.api!.deleteSync(byId: id))
+            }
+        }
+
+        /**
+         Delete a sync
+         
+         ```
+         syncManager.Promises.deleteSync(name:  name)
+         .then {
+         ..
+         }
+         ```
+         - parameter name: Name of sync.
+         */
+        func deleteSync(name: String) -> Promise<Void>  {
+            return Promise(.pending) {  resolver in
+                resolver.fulfill(self.api!.deleteSync(byName:name))
+            }
+        }
+
+        /**
+         Delete a sync
+         
+         ```
+         syncManager.Promises.createSyncDown(target: target, options: options, soupName: soupName, syncName: syncName)
+         .then {
+         ..
+         }
+         ```
+         - parameters
+            - target: SFSyncDownTarget target
+            - options: SFSyncOptions
+             - soupName: Soup Name
+             - syncName: Sync Name
+         - Returns: SFSyncState wrapped in a promise.
+         */
+        func createSyncDown(target: SFSyncDownTarget, options: SFSyncOptions, soupName: String, syncName: String?) -> Promise<SFSyncState> {
+            return Promise(.pending) {  resolver in
+                resolver.fulfill( self.api!.createSyncDown(target, options: options, soupName: soupName, syncName: syncName)
+            }
+        }
+
+        /**
+         Sync Down
+         
+         ```
+         syncManager.Promises.deleteSync(id:  id)
+         .then {
+         ..
+         }
+         ```
+         - parameters
+             - target: SFSyncDownTarget target
+             - soupName: Soup Name
+             - updateBlock: Block to invoke
+         - Returns: SFSmartStore wrapped in a promise.
+         */
+        func syncDown(target: SFSyncDownTarget, soupName: String, updateBlock: SFSyncSyncManagerUpdateBlock) -> Promise<SFSyncState> {
+            return Promise(.pending) {  resolver in
+                resolver.fulfill(self.api!.syncDown(with: target, soupName: soupName, update: updateBlock))
```

Comment:
The sync down call returns right away - should the promise resolve when the sync actually completes.
So we could do something similar to what we did in the react native bridge: https://github.com/forcedotcom/SalesforceMobileSDK-iOS/blob/master/libs/SalesforceReact/SalesforceReact/Classes/SFSmartSyncReactBridge.m#L97